ui/src/criteria/attribute.tsx

```diff
@@ -131,6 +155,127 @@ class _ implements CriteriaPlugin<Data> {
+  const [maxValue, setMaxValue] = useState(range.max || maxBound);
+
+  const updateValue = (newMin: number, newMax: number) => {
+    setMinValue(newMin);
+    setMaxValue(newMax);
+    dispatchFn(
+      produce(data, (oldData) => {
+        oldData.dataRanges[index].min = newMin;
+        oldData.dataRanges[index].max = newMax;
+      })
+    );
+  };
+
+  const handleChange = (event: Event, newValue: number | number[]) => {
+    const [newMin, newMax] = newValue as number[];
+    setMinInputValue(String(newMin));
+    setMaxInputValue(String(newMax));
+    updateValue(newMin, newMax);
+  };
+
+  // Make sure empty input won't get changed.
+
+  const handleMinInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setMinInputValue(event.target.value);
+    const newMin = event.target.value === "" ? 0 : Number(event.target.value);
+    updateValue(Math.min(maxValue, Math.max(minBound, newMin)), maxValue);
+  };
+  const handleMinInputBlur = () => {
+    setMinInputValue(String(minValue));
+  };
+
+  const handleMaxInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setMaxInputValue(event.target.value);
+    const newMax = event.target.value === "" ? 0 : Number(event.target.value);
+    updateValue(Math.max(minValue, Math.min(maxBound, newMax)), minValue);
+  };
+  const handleMaxInputBlur = () => {
+    setMaxInputValue(String(maxValue));
+  };
+
+  const handleDeleteRange = () => {
+    dispatchFn(
+      produce(data, (oldData) => {
+        oldData.dataRanges.splice(index, 1);
+      })
+    );
+  };
+
+  return (
+    <Box sx={{ width: "30%", minWidth: 500, margin: 5 }}>
+      <Grid container spacing={2} direction="row" alignItems="center">
+        <Grid item>
+          <Input
+            value={minInputValue}
+            size="medium"
+            onChange={handleMinInputChange}
+            onBlur={handleMinInputBlur}
+            inputProps={{
+              step: Math.ceil((maxBound - minBound) / 20),
+              min: minBound,
+              max: maxBound,
+              type: "number",
+              "aria-labelledby": "input-slider",
+            }}
+          />
+        </Grid>
+        <Grid item xs>
+          <Slider
+            value={[minValue, maxValue]}
+            onChange={handleChange}
+            valueLabelDisplay="auto"
+            getAriaValueText={(value) => value.toString()}
+            min={minBound}
+            max={maxBound}
+            disableSwap
+          />
+        </Grid>
+        <Grid item>
+          <Input
+            value={maxInputValue}
+            onChange={handleMaxInputChange}
+            onBlur={handleMaxInputBlur}
+            inputProps={{
+              step: Math.ceil((maxBound - minBound) / 20),
+              min: minBound,
+              max: maxBound,
+              type: "number",
+              "aria-labelledby": "input-slider",
+            }}
+          />
+        </Grid>
+        <DeleteIcon
```

Comment:
Use a [IconButton](https://mui.com/material-ui/react-button/#icon-button) instead of just an icon. It will worry about things like the cursor for you.

Reply:
Done